scopes/lanes/ui/hooks/lanes-context/lanes-model.ts

```diff
@@ -0,0 +1,117 @@
+import { ComponentModel } from '@teambit/component';
+import { ScopeModel } from '@teambit/scope.models.scope-model';
+import { ComponentID, ComponentIdObj } from '@teambit/component-id';
+
+export type LaneComponentQueryResult = {
+  id: ComponentIdObj;
+  head: string;
+};
+export type LaneQueryResult = {
+  name: string;
+  remote?: string;
+  isMerged: boolean;
+  components: LaneComponentQueryResult[];
+};
+export type LanesQueryResult = {
+  lanes?: {
+    getLanes?: LaneQueryResult[];
+    getCurrentLaneName?: string;
+  };
+};
+export type LanesHost = 'workspace' | 'scope';
+export type LaneComponentModel = { model: ComponentModel; url: string };
+export type LaneModel = {
+  id: string;
+  scope: string;
+  url: string;
+  name: string;
+  isMerged: boolean | null;
+  components: LaneComponentModel[];
+};
+export type LanesModelProps = {
+  lanes: LaneModel[];
+  currentLane?: LaneModel;
+};
+
+export class LanesModel {
+  static baseLaneRoute = '/~lane';
+
+  static laneRouteUrlRegex = `${LanesModel.baseLaneRoute}/:orgId([\\w-]+)/:scopeId([\\w-]+)/:laneId([\\w-]+)`;
+
+  static laneComponentIdUrlRegex = '[\\w\\/-]*[\\w-]';
+  static laneComponentUrlRegex = `${LanesModel.laneRouteUrlRegex}/:componentId(${LanesModel.laneComponentIdUrlRegex})`;
+
+  static getLaneUrl = (laneId: string) => `${LanesModel.baseLaneRoute}/${laneId.replace('.', '/')}`;
+  static getLaneComponentUrl = (componentId: ComponentID, laneId?: string) =>
+    laneId ? `${LanesModel.getLaneUrl(laneId)}/${componentId.fullName}?version=${componentId.version}` : '';
+
+  static mapToLaneModel(laneData: LaneQueryResult, currentScope: ScopeModel): LaneModel {
+    const { name, remote, isMerged } = laneData;
+    const laneName = name;
+    const laneId = remote || name;
+    const laneScope = remote?.split('/')[0] || currentScope.name;
+
+    const components = laneData.components.map((component) => {
+      const componentModel = ComponentModel.from({
+        id: { ...component.id, version: component.head, scope: component.id.scope || laneScope },
+        displayName: component.id.name,
+        compositions: [],
+        packageName: '',
+        description: '',
+      });
+      return { model: componentModel, url: LanesModel.getLaneComponentUrl(componentModel.id, laneId) };
+    });
+
+    return {
+      id: laneId,
+      name: laneName,
+      scope: laneScope || currentScope.name,
+      isMerged,
+      url: LanesModel.getLaneUrl(laneId),
+      components,
+    };
+  }
+
+  static groupByScope(lanes: LaneModel[]): Map<string, LaneModel[]> {
+    return lanes.reduce((accum, next) => {
```

Comment:
please don't use reduce, it makes the code less readable :)

a more simple way would be
```
const grouped = new Map...();
lanes.forEach(x=> grouped.set(...))
return grouped;
```